tools/run_tests/run_tests.py

```diff
@@ -884,6 +884,42 @@ def runs_per_test_type(arg_str):
         msg = '\'{}\' is not a positive integer or \'inf\''.format(arg_str)
         raise argparse.ArgumentTypeError(msg)
 
+_COMPILER_CHOICES = ('default',
+                     'gcc4.4', 'gcc4.6', 'gcc4.9', 'gcc5.3',
+                     'clang3.4', 'clang3.5', 'clang3.6', 'clang3.7',
+                     'vs2010', 'vs2013', 'vs2015',
+                     'python2.7', 'python3.4',
+                     'node0.12', 'node4', 'node5',
+                     'coreclr',)
+
+
+def _is_travis_enabled():
+  for arg in sys.argv:
+    if arg == "--travis" or arg == "-t":
+      return True
+  return False
+
+
+# TODO (https://github./grpc/grpc/issues/7934): This workaround would not
+# have been necessary if the testing infrastructure provided provisions for
+# supporting different configurations for different branches on the repository.
+# Remove this workaround once test infrastructure supports configurations for
+# different branches.
+def _compiler_choices(compiler):
+  if _is_travis_enabled():
+    if compiler in _COMPILER_CHOICES:
+      return compiler
+    else:
+      return 'python2.7'
```

Comment:
So, this is a bit more complex than that, because this should be used by more than just Python.

We rather need to return "none" or "freepass" or "void" or whatever wording you think is appropriate. Then later, in the _build_and_run function, we need to check for this, and build nothing and test nothing, but return a valid status code, and an empty junit test file (the xml_report basically).